internal/cmd/oohelperd/internal/nwcth/dns.go

```diff
@@ -0,0 +1,39 @@
+package nwcth
+
+import (
+	"context"
+	"net"
+
+	"github.com/apex/log"
+	"github.com/ooni/probe-cli/v3/internal/engine/experiment/nwebconnectivity"
+	"github.com/ooni/probe-cli/v3/internal/engine/netx"
+	"github.com/ooni/probe-cli/v3/internal/engine/netx/archival"
+)
+
+// newfailure is a convenience shortcut to save typing
+var newfailure = archival.NewFailure
+
+// CtrlDNSResult is the result of the DNS check performed by
+// the Web Connectivity test helper.
+type CtrlDNSResult = nwebconnectivity.ControlDNS
+
+// DNSConfig configures the DNS check.
+type DNSConfig struct {
+	Domain string
+}
+
+// NewResolver creates a new DNS resolver instance
+func NewResolver() netx.Resolver {
```

Comment:
Unless there's a reason to export `NewResolver`, I think we can rename it `newResolver`.